runtime/caml/mlvalues.h

```diff
@@ -373,6 +373,13 @@ CAMLextern header_t *caml_atom_table;
 #define Val_emptylist Val_int(0)
 #define Tag_cons 0
 
+/* Option constructors */
+
+#define Val_none Val_int(0)
+CAMLextern value caml_Val_some(value v);
+#define Val_some(v) caml_Val_some(v)
+#define Some_val(v) Field(v,0)
```

Comment:
At first I was unhappy with the naming `Some_val`, as the convention in our C macros `Foo_val` is that they return a usable C value at the type `foo`. In fact there are a handful of cases following the present approach (where `Foo` is an accessor that returns an OCaml value), namely `Forward_val` and `Class_val`.

Personally I would still have a preference for some other name, such as `Some_field(v)` (there is no index/offset parameter because there is a single field).

Reply:
After thinking about this a bit more: there is a symmetry between `Val_some` and `Some_val` which explains the naming. I guess the current name is fine (yet unpleasant).